awsbestpractices/missing-pagination/MissingPagination.java

```diff
@@ -0,0 +1,507 @@
+        conditions.put("STATUS",stateCondition);
+
+        ScanRequest scanRequest = new ScanRequest()
+                .withTableName(tableName)
+                .withScanFilter(conditions)
+                .withAttributesToGet(Arrays.asList("STORE_ID"));
+
+
+        ScanResult result = dynamoClient.scan(scanRequest);
+        if(result==null || result.getScannedCount()==0){
+            logger.info("zero scanned");
+        }
+
+        logger.info("Set of expired stores found");
+
+        for (Map<String, AttributeValue> item : result.getItems()) {
+
+            String storeID = item.get("STORE_ID").getS();
+            expiredStores.add(storeID);
+
+        }
+
+        return expiredStores;
+
+    }
+
+    // Missing
+    // Adapted from
+    // https://code.amazon.com/packages/AwsGlueMlArtifactsDeletionLambda/blobs/0b683300e7a9e42b10a7392a52a9e7f3d64308b2/--/src/com/amazon/glue/ml/lambda/artifacts/deletion/OldMLTransformArtifactsDeletion.java#L128-L133
+    public Set<String> missingPagination_2(AmazonS3 s3Client, TransformCleanupEntry transformCleanupEntry) {
+        MLModelArtifact mlModelArtifact = new MLModelArtifact(transformCleanupEntry.getAccountId(), transformCleanupEntry.getTransformId(),
+                "dummyFileName");
+
+        String bucketName = mlModelArtifact.bucketName(stage, region);
+        String key = String.format("%s/%s/", transformCleanupEntry.getAccountId(), transformCleanupEntry.getTransformId());
+        List<S3ObjectSummary> s3ObjectSummaries = s3Client.listObjects(bucketName, key).getObjectSummaries();
+        List<DeleteObjectsRequest.KeyVersion> keys = new ArrayList<>();
+        for (S3ObjectSummary s3ObjectSummary : s3ObjectSummaries) {
+            log.info("Deleting file : {} with bucketName : {}", s3ObjectSummary.getKey(), bucketName);
+            keys.add(new DeleteObjectsRequest.KeyVersion(s3ObjectSummary.getKey()));
+        }
+        if (!keys.isEmpty()) {
+            s3Client.deleteObjects(new DeleteObjectsRequest(bucketName).withKeys(keys));
+        }
+    }
+
+    // Missing
+    // Adapted from
+    // https://code.amazon.com/packages/AWSServerlessAppsRepoStorage/blobs/6c58cdb1b0bffb657cea856afb1f450ef10a6003/--/src/com/amazonaws/serverlessappsrepo/storage/S3StorageService.java#L323-L341
+    public Set<String> missingPagination_3(final AmazonS3 s3Client, final String bucketName, final String prefix) {
+        ListObjectsV2Request listObjectsV2Request = new ListObjectsV2Request()
+                .withBucketName(bucketName)
+                .withPrefix(prefix);
+
+        ListObjectsV2Result listObjectsV2Result;
+
+        try (Timer.Context s3ListObjectsTimer = MetricBuilder.of(getClass(), "s3ListObjectsAPI.totalTime")
+                .timer()
+                .time()) {
+            log.info("List s3 objects.. bucket: {}, prefix: {}", bucketName, prefix);
+            listObjectsV2Result = s3Client.listObjectsV2(listObjectsV2Request);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.